mne/preprocessing/_fine_cal.py

```diff
@@ -3,9 +3,421 @@
 
 # License: BSD (3-clause)
 
+from collections import defaultdict, OrderedDict
+from functools import partial
+
 import numpy as np
 
-from ..utils import check_fname, _check_fname
+from ..bem import _check_origin
+from ..io.pick import pick_info, pick_types
+from ..io import _loc_to_coil_trans, _coil_trans_to_loc, BaseRaw
+from ..transforms import _find_vector_rotation
+from ..utils import (logger, verbose, check_fname, _check_fname, _pl,
+                     _ensure_int, _check_option, _validate_type, _reg_pinv)
+
+from .maxwell import (_col_norm_pinv, _trans_sss_basis, _prep_mf_coils,
+                      _get_grad_point_coilsets, _read_cross_talk,
+                      _prep_fine_cal)
+
+
+@verbose
+def compute_fine_calibration(raw, n_imbalance=3, t_window=10., ext_order=2,
+                             origin=(0., 0., 0.), cross_talk=None,
+                             fine_cal=None, verbose=None):
+    """Compute fine calibration from empty-room data.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data to use. Should be from an empty-room recording,
+        and all channels should be good.
+    n_imbalance : int
+        Can be 1 or 3 (default), indicating the number of gradiometer
+        imbalance components. Only used if gradiometers are present.
+    t_window : float
+        Time window to use for surface normal rotation in seconds.
+        Default is 10.
+    %(maxwell_ext)s
+        Default is 2, which is lower than the default (3) for
+        :func:`mne.preprocessing.maxwell_filter` because it tends to yield
+        more stable parameter estimates.
+    %(maxwell_origin)s
+    %(maxwell_cross)s
+    fine_cal : dict | None
+        Dictionary with existing calibration. If provided, the magnetometer
+        imbalances and adjusted normals will be used and only the gradiometer
+        imbalances will be estimated (see step 2 in Notes below).
+    %(verbose)s
+
+    Returns
+    -------
+    fine_cal : dict
+        Fine calibration data.
+    count : int
+        The number of good segments used to compute the magnetometer
+        parameters.
+
+    See Also
+    --------
+    mne.preprocessing.maxwell_filter
+
+    Notes
+    -----
+    This algorithm proceeds in two steps, both optimizing the fit between the
+    data and a reconstruction of the data based only on an external multipole
+    expansion:
+
+    1. Estimate magnetometer normal directions and scale factors. All
+       coils (mag and matching grad) are rotated by the adjusted normal
+       direction.
+    2. Estimate gradiometer imbalance factors. These add point magnetometers
+       in just the gradiometer difference direction or in all three directions
+       (depending on ``n_imbalance``).
+
+    Step (1) is typically the most time consuming step, so step (2) can be
+    iteratively performed using 1 or 3 directions by passing ``calibration``
```

Comment:
```suggestion
    iteratively performed using 1 or 3 directions by passing to ``calibration``
```

? I don't get the sentence